src/smart-components/role/edit-resource-definitions-modal.js

```diff
@@ -0,0 +1,193 @@
+};
+
+const EditResourceDefinitionsModal = ({ cancelRoute }) => {
+  const routeMatch = useRouteMatch(paths['role-detail-permission-edit']);
+  const {
+    params: { permissionId, roleId },
+  } = useRouteMatch(routeMatch);
+  const { replace, push } = useHistory();
+
+  const dispatch = useDispatch();
+  const fetchResourceDefinitions = () => dispatch(getResourceDefinitions());
+
+  const [resourcesPath, setResourcesPath] = useState(undefined);
+  const [cancelWarningVisible, setCancelWarningVisible] = useState(false);
+  const [loadingStateVisible, setLoadingStateVisible] = useState(true);
+  const [changedResources, setChangedResources] = useState();
+
+  const { definedResources, role } = useSelector(
+    (state) => ({
+      role: state.roleReducer.selectedRole,
+      definedResources: state.roleReducer.selectedRole?.access
+        ? state.roleReducer.selectedRole.access
+            .find((a) => a.permission === permissionId)
+            .resourceDefinitions.reduce((acc, curr) => [...acc, curr.attributeFilter.value], []) // tady to jsem přehodil komentář, nefunguje!
+        : [],
+      isRecordLoading: state.roleReducer.isRecordLoading,
+    }),
+    shallowEqual
+  );
+
+  const { resourceTypes, isLoading, isLoadingResources, resources } = useSelector((props) => selector(props, resourcesPath), shallowEqual);
+
+  useEffect(() => {
+    fetchResourceDefinitions();
+  }, [permissionId]);
+
+  useEffect(() => {
+    if (!isLoading) {
+      let path = resourceTypes.find((r) => r.value === permissionId.split(':')?.[1])?.path;
+      if (path) {
+        setResourcesPath(path.split('/')[5]);
+        dispatch(getResource(path));
+      } else if (permissionId.split(':')?.[1] === '*') {
+        setResourcesPath('*');
+        resourceTypes.map((r) => dispatch(getResource(r.path)));
+      }
+    }
+  }, [resourceTypes]);
+
+  const onCancel = () => replace(cancelRoute);
+
+  const handleCancel = (data) => {
+    if (data['dual-list-select'] === definedResources) {
+      onCancel();
+    } else {
+      setChangedResources(data['dual-list-select']);
+      setCancelWarningVisible(true);
+    }
+  };
+
+  const handleSubmit = (data, options) => {
+    setChangedResources(data['dual-list-select']);
+    const newAccess = {
+      permission: permissionId,
+      resourceDefinitions: data['dual-list-select'].map((value) => ({
+        attributeFilter: {
+          key: `cost-management.${options.find((option) => option.value === value).path}`,
+          operation: 'equal',
+          value,
+        },
+      })),
+    };
+    dispatch(updateRole(roleId, { ...role, access: role.access.map((item) => (item.permission === permissionId ? newAccess : item)) }), true).then(
+      () => {
+        dispatch(fetchRole(roleId));
+        push(cancelRoute);
+      }
+    );
+  };
+
+  const options = createOptions(resources);
+
+  return (
+    <React.Fragment>
+      <WarningModal
```

Comment:
No need to wrap in `React.Fragment`

Reply:
I think it's needed since I have 2 elements - WarningModal and Modal/FormRenderer below